pandas/_libs/lib.pyx

```diff
@@ -465,7 +465,11 @@ cpdef ndarray[object] astype_unicode(ndarray arr):
     for i in range(n):
         # we can use the unsafe version because we know `result` is mutable
         # since it was created from `np.empty`
-        util.set_value_at_unsafe(result, i, unicode(arr[i]))
+        arr_i = arr[i]
+        util.set_value_at_unsafe(
+            result,
+            i,
+            unicode(arr_i) if arr_i is not np.nan else np.nan)
```

Comment:
Interesting spacing...maybe we should do this instead:

~~~
uni_arr_i = unicode(arr_i) if arr_i is not np.nan else np.nan
util.set_value_at_unsafe(result, i, uni_arr_i)
~~~

Reply:
use ``np.isnan`` here

Reply:
Using `np.nan` here raises:
```
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
When i use that, all hell breaks loose. I get errors in tests like this one https://github.com/pandas-dev/pandas/blob/master/pandas/tests/frame/test_dtypes.py#L533

Is it because they use `np.NaN`? It looks like `checknull` checks both `np.NaN` and `np.nan`, while before the change I used to check only `np.nan`.
If that's the case, then I have to modify more tests.

Reply:
@gfyoung are you sure this indentation is a big problem? Because if I do what you suggest, then how should I declare uni_arr_i (and str_arr_i) in the cdef? 
Would it be ok if I changed it to sth like
```python
util.set_value_at_unsafe(
    ...
)
```
(moved the close bracket in the next line)?

Reply:
That would work as well.

Reply:
the nans are the same; iow they point to the same object
go ahead and change tests if need be and i will have a look